apps/condo/domains/news/components/RecipientCounter.tsx

```diff
@@ -243,44 +269,94 @@ export const RecipientCounter: React.FC<RecipientCounterProps> = ({ newsItemScop
+                    isLoading={isLoadingCounters}
+                />
+            </Col>
+            <Col>
+                <Counter
+                    label={WillNotReceiveLabelMessage}
+                    value={unitsCount - receiversCount}
+                    type='danger'
+                    hint={willNotReceiveUnitsCount === 0 ? WillZeroNotReceiveHintMessage : formatWillNotReceiveHintMessage(willNotReceiveUnitsCount)}
+                    downloadButton={(
+                        <NewsItemRecipientsExportToXlsxButton key='exportToExcel' />
+                    )}
+                    isLoading={isLoadingCounters}
+                />
+            </Col>
+        </RecipientCounterContainer>
     )
 }
+
+export const NewsItemSharingCustomRecipientCounter: React.FC<{ contextId: string, newsItemScopes: NewsItemScopeNoInstanceType[] }> = ({ contextId, newsItemScopes }) => {
+    const [counter, setCounter] = useState(null)
+
+    const intl = useIntl()
+    const MailingMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.mailing' })
+    const WillReceiveLabelMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.willReceive.label' })
+    const ErrorLoadingMessage = intl.formatMessage({ id: 'news.component.RecipientCounter.error.loading' })
+
+    const [getCounter, { loading: isCounterLoading }] = useLazyQuery(
+        GET_NEWS_SHARING_RECIPIENTS_COUNTERS_QUERY,
+        {
+            onCompleted: (data) => {
+                console.log(data)
+                console.log(get(data, ['result', 'receiversCount']))
+                setCounter(get(data, ['result', 'receiversCount']))
+            },
+            onError: (error) => {
+                console.error({ msg: 'Failed to load recipients counters', error })
+                const message = get(error, ['graphQLErrors', 0, 'extensions', 'messageForUser'], ErrorLoadingMessage)
+                notification.error({ message })
+            },
+            fetchPolicy: 'cache-first',
+        },
+    )
+
+    const processedNewsItemScope = useMemo(() => {
+        return processNewsItemScopes(newsItemScopes)
+    }, [newsItemScopes])
+
+    useEffect(() => {
+        const sender = getClientSideSenderInfo()
+        const meta = { dv: 1, sender }
+
+        getCounter({
+            variables: {
+                data: {
+                    newsItemScopes: processedNewsItemScope,
+                    b2bAppContext: { id: contextId },
+                    ...meta,
+                },
+            },
+        })
+    }, [getCounter, contextId, newsItemScopes])
+
+    // if typeof counter !== number is used here instead of just if !counter because bool(0) => false
+    const isLoadingCounter = isCounterLoading || typeof counter !== 'number'
+
+    return (
+        <RecipientCounterContainer
+            title={<>{MailingMessage}<Typography.Text strong>{buildMessageFromNewsItemScopes(newsItemScopes, intl)}</Typography.Text></>}
```

Comment:
it looks hacky. is it possible to move inise `RecipientCounterContainer` or create a special component?